src/gui/transferlistmodel.cpp

```diff
@@ -663,17 +666,14 @@ BitTorrent::Torrent *TransferListModel::torrentHandle(const QModelIndex &index)
 
 void TransferListModel::handleTorrentAboutToBeRemoved(BitTorrent::Torrent *const torrent)
 {
-    const int row = m_torrentMap.value(torrent, -1);
+    const qsizetype row = m_torrentMap.value(torrent, -1);
     Q_ASSERT(row >= 0);
 
     beginRemoveRows({}, row, row);
     m_torrentList.removeAt(row);
     m_torrentMap.remove(torrent);
-    for (int &value : m_torrentMap)
-    {
-        if (value > row)
-            --value;
-    }
+    for (qsizetype i = row; i < m_torrentList.size(); ++i)
```

Comment:
Why not range based `for`?

Reply:
how?

Reply:
Won't something like the following work?
```c++
    for (Torrent *torrent : asConst(m_torrentList))
        --m_torrentMap[torrent];
```

Reply:
No, that would be wrong. Note that here `i = row;` not `i = 0;`

Reply:
> Note that here `i = row;` not `i = 0;`

Oops... I misread it.

Reply:
But why did you change it at all? Wasn't the previous way more effective? Iterating the QHash should be much less overhead compared to repeatedly lookup value by key.

Reply:
>Wasn't the previous way more effective? 

It depends. If `row` is near to the end then this new iteration will be faster.
IMO there should be a cost function to decide which method is faster but I find it hard to define one.
Here is a naive attempt:
```c++
    // find which method is faster
    const int comparisonCost = 50 / 3;  // ns, assuming: 50 cpu cycles for integer comparison, cpu clock is 3 GHz
    const int memoryAccessCost = 100; // ns, https://stackoverflow.com/a/4087315
    const int IterateAllTime = m_torrentList.size() * (memoryAccessCost + comparisonCost);
    const int shiftTime = (m_torrentList.size() - row) * (2 * memoryAccessCost);

    if (shiftTime < IterateAllTime)
    {
        for (qsizetype i = row; i < m_torrentList.size(); ++i)
            --m_torrentMap[m_torrentList[i]];
    }
    else
    {
        for (int &value : m_torrentMap)
        {
            if (value > row)
                --value;
        }
    }
```
The `memoryAccessCost` is controversial as it might be "CPU L1/L2  CACHE reference" which is 5/7 ns instead of 100 ns.

Reply:
Changed to a simpler (approximated) cost model.